core/tests/widgets/test_slider.py

```diff
@@ -1,203 +1,469 @@
+        ):
+            slider.range = (min, max)
+
+
+def test_set_enabled_with_working_values(slider, on_change):
+    assert slider.enabled == INITIAL_ENABLED
+    slider.enabled = False
+    assert slider.enabled is False
+
+
+def test_get_tick_count(slider, on_change):
+    tick_count = slider.tick_count
+    assert INITIAL_TICK_COUNT == tick_count
+
+
+TICK_RANGE = (10, 110)
+TICK_PARAM_NAMES = "tick_count, tick_step, tick_value, value"
+TICK_PARAM_VALUES = [
+    (None, None, None, 50),
+    (11, 10, 5, 50),
+    (5, 25, 3, 60),  # Rounded up
+    (4, 33.33333, 2, 43.33333),  # Rounded down
+    (2, 100, 1, 10),  # Rounded down to the minimum
+]
+
+
+@pytest.mark.parametrize(TICK_PARAM_NAMES, TICK_PARAM_VALUES)
+def test_set_tick_count(slider, on_change, tick_count, tick_step, tick_value, value):
+    slider.range = TICK_RANGE
+    slider.tick_count = tick_count
+    assert slider.tick_count == tick_count
+    assert attribute_value(slider, "tick_count") == tick_count
+
+    if tick_step is None:
+        assert slider.tick_step is None
+    else:
+        assert isinstance(slider.tick_step, float)
+        assert slider.tick_step == approx(tick_step)
+
+    assert_value(
+        slider,
+        on_change,
+        approx(value),
+        tick_value=tick_value,
+        change_count=0 if (value == INITIAL_VALUE) else 1,
+    )
+
+
+@pytest.mark.parametrize(TICK_PARAM_NAMES, TICK_PARAM_VALUES)
+def test_set_value_with_tick_count(
+    slider, on_change, tick_count, tick_step, tick_value, value
+):
+    slider.range = TICK_RANGE
+    slider.tick_count = tick_count
+    slider.value = TICK_RANGE[1]
+
+    on_change.reset_mock()
+    slider.value = INITIAL_VALUE
+    assert_value(
+        slider, on_change, approx(value), tick_value=tick_value, change_count=1
+    )
+
+    # Resetting the same value should round to the same result, so on_change should not be
+    # called.
+    on_change.reset_mock()
+    slider.value = INITIAL_VALUE
+    assert_value(
+        slider, on_change, approx(value), tick_value=tick_value, change_count=0
+    )
+
+
+def test_set_tick_count_too_small(slider, on_change):
+    for tick_count in [1, 0, -1]:
+        with raises(ValueError, match="tick count must be at least 2"):
+            slider.tick_count = tick_count
+
+
+def test_focus(slider, on_change):
+    slider.focus()
+    assert_action_performed(slider, "focus")
+
+
+def test_set_on_change():
+    on_change = Mock()
+    slider = toga.Slider(on_change=on_change)
+    assert slider.on_change._raw == on_change
+
+
+def test_set_on_press():
+    on_press = Mock()
+    slider = toga.Slider(on_press=on_press)
+    assert slider.on_press._raw == on_press
+
+
+def test_set_on_release():
+    on_release = Mock()
+    slider = toga.Slider(on_release=on_release)
+    assert slider.on_release._raw == on_release
+
+
+def assert_value(slider, on_change, value, *, tick_value=None, change_count=0):
```

Comment:
This assertion does a lot of heavy lifting in the tests - it would be good to include a docstring summarising what it is asserting.

Reply:
Done.